### ISSUE TITLE
Still: Could not find field com.sun.beans.WeakCache.map on class com.sun.beans.WeakCache

### ISSUE THREAD

**Issue by ppalaga** [quarkusio/quarkus#2347]
The fix for https://github.com/quarkusio/quarkus/issues/2207 is not perfect, still getting the following error when building `integration-tests/camel-core` with `-Dnative-camel`:

```
[INFO] --- quarkus-maven-plugin:999-SNAPSHOT:native-image (native-image) @ quarkus-integration-test-camel-core ---
[INFO] [io.quarkus.creator.phase.nativeimage.NativeImagePhase] Running Quarkus native-image plugin on OpenJDK 64-Bit Server VM
[INFO] [io.quarkus.creator.phase.nativeimage.NativeImagePhase] /home/ppalaga/bin/graalvm/graalvm-ce-1.0.x/bin/native-image -J-Djava.util.logging.manager=org.jboss.logmanager.LogManager -J-Dio.netty.leakDetection.level=DISABLED -J-Dcom.sun.xml.bind.v2.bytecode.ClassTailor.noOptimize=true -J-DCamelSimpleLRUCacheFactory=true -J-Dio.netty.noUnsafe=true -H:InitialCollectionPolicy=com.oracle.svm.core.genscavenge.CollectionPolicy$BySpaceAndTime -jar quarkus-integration-test-camel-core-999-SNAPSHOT-runner.jar -J-Djava.util.concurrent.ForkJoinPool.common.parallelism=1 -H:FallbackThreshold=0 -H:-AddAllCharsets -H:EnableURLProtocols=http -H:NativeLinkerOption=-no-pie -H:-SpawnIsolates -H:-JNI --no-server -H:-UseServiceLoaderFeature -H:+StackTrace
[quarkus-integration-test-camel-core-999-SNAPSHOT-runner:24689]    classlist:   8,610.37 ms
14:02:51,510 INFO  [com.sun.xml.bin.v2.run.ref.opt.AccessorInjector] The optimized code generation is disabled
[quarkus-integration-test-camel-core-999-SNAPSHOT-runner:24689]        (cap):     911.02 ms
[quarkus-integration-test-camel-core-999-SNAPSHOT-runner:24689]        setup:   2,263.87 ms
14:02:54,026 INFO  [org.apa.cam.imp.con.FastTypeConverterRegistry] Type converters loaded (core: 183, classpath: 34)
14:02:54,304 INFO  [org.jbo.res.res.i18n] RESTEASY002225: Deploying javax.ws.rs.core.Application: class io.quarkus.it.camel.core.CamelApplication
14:02:55,068 INFO  [org.jbo.threads] JBoss Threads version 3.0.0.Beta3
14:02:55,482 INFO  [org.xnio] XNIO version 3.7.0.Final
14:02:55,532 INFO  [org.xni.nio] XNIO NIO Implementation Version 3.7.0.Final
[quarkus-integration-test-camel-core-999-SNAPSHOT-runner:24689]     analysis:   9,704.80 ms
Error: should not reach here: Could not find field com.sun.beans.WeakCache.map on class com.sun.beans.WeakCache
        object com.sun.beans.WeakCache
        object com.sun.beans.finder.PropertyEditorFinder
        object java.beans.ThreadGroupContext
        object java.beans.WeakIdentityMap$Entry
        object java.beans.WeakIdentityMap$Entry[]
        object java.beans.ThreadGroupContext$1
        method java.beans.ThreadGroupContext.getContext()
Error: Use -H:+ReportExceptionStackTraces to print stacktrace of underlying exception
```

Adding `@Substitute` on `io.quarkus.camel.core.runtime.graal.Target_com_sun_beans_WeakCache.map` seems to solve the issue.

A PR follows.

**Comment by gnodet** [comment]
Did you create a PR with the fix https://github.com/ppalaga/quarkus/commit/eceeb8684489b8c30bb28bc595e7d10bb4bd7302 ?

**Comment by ppalaga** [comment]
> Did you create a PR with the fix [ppalaga@eceeb86](https://github.com/ppalaga/quarkus/commit/eceeb8684489b8c30bb28bc595e7d10bb4bd7302) ?

Yes, it is a part of this one: https://github.com/quarkusio/quarkus/pull/2352 where I am still solving `Error: Class that is marked for delaying initialization to run time got initialized during image building: com.sun.xml.internal.bind.v2.runtime.reflect.opt.Injector`
